pkg/dynamic/discovery/discovery.go

```diff
@@ -99,8 +99,11 @@ func (rm *ResourceMap) refresh() {
 	logging.Logger.V(7).Info("Refreshing API discovery info")
 	_, groups, err := rm.discoveryClient.ServerGroupsAndResources()
 	if err != nil {
-		logging.Logger.Error(err, "Failed to fetch discovery info")
-		return
+		if len(groups) == 0 {
+			logging.Logger.Error(err, "Failed to fetch discovery info")
+			return
+		}
+		logging.Logger.V(4).Info("Failed to fetch all resources, continuing with partial discovery info", "failures", err)
```

Comment:
@AmitKumarDas I updated the log level, the message looks like this now:
```
{"level":"Level(-4)","ts":1638799811.1609228,"msg":"Failed to fetch all resources, continuing with partial discovery info","failures":"unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request"}
```
The only thing I wasn't sure was the name of the key, so I went with "failures".

Reply:
Thanks. Looks good.